internal/controller/driver_controller.go

```diff
@@ -313,6 +317,24 @@ func (r *driverReconcile) LoadAndValidateDesiredState() error {
 		return err
 	}
 
+	// check if the driver wants CSI-Addons features
+	if feature := r.driver.GetAnnotations()[driverCSIAddonsFeatureVolumeCondition]; feature != "" {
+		enabled, err := strconv.ParseBool(feature)
+		if err != nil {
+			r.log.Error(
+				err,
+				"Unable to parse annotation on driver.csi.ceph.io",
+				"name",
+				client.ObjectKeyFromObject(&r.driver),
+				driverCSIAddonsFeatureVolumeCondition,
+				feature,
+			)
+			return err
+		}
+
+		r.withCsiAddonsVolumeCondition = enabled
+	}
```

Comment:
This check can be moved to inside reconcileNodePluginDeamonSetForCsiAddons we dont need a new variable as well `withCsiAddonsVolumeCondition`, please add a check for this only if the driver is cephFS

Reply:
Well, even for RBD the feature can be enabled. By default RBD will have the standard CSI-Addons DaemonSet, but with the annotation also the volume condition of RBD volumes will be reported.

I can move it into `reconcileNodePluginDeamonSetForCsiAddons`, and check for `!r.isNfsDriver()` to prevent issues with it there.

Does that make sense to you too, @Madhu-1 ?

Reply:
@nixpanic it wont work for RBD right, because its missing RBAC's isnt it? if its enabled and works that should be fine.